docker/start.py

```diff
@@ -59,6 +59,18 @@ def generate_secrets(environ, secrets):
         if not os.path.exists("/compiled"): os.mkdir("/compiled")
 
         config_path = "/compiled/homeserver.yaml"
+        
+        # Convert SYNAPSE_NO_TLS to boolean if exists
+        if "SYNAPSE_NO_TLS" in environ:
+            tlsanswerstring = str.lower(environ["SYNAPSE_NO_TLS"])
```

Comment:
might have been nice to write `tlsanswerstring = str.lower(environ.get("SYNAPSE_NO_TLS", "false"))` here, to avoid the `if` condition above.